### DISCUSSION TITLE
希望增加对visionOS的支持

### DISCUSSION THREAD

**Discussion by wangdinglu** [Tencent/puerts#1613] [Q&A]
求教有没有办法可以支持visionOS的Build，本来希望直接用iOS的就可以，但实测不太行会报FAQ第一个的Bug，不知道是设置问题还是需要重新编版本，希望有大佬可以解答一下或者官方可以支持 orzzzzzzzzzzzz

**Comment by chexiongsheng** [comment]
FAQ第一个bug是什么?unity还是ue？

**Comment by chexiongsheng** [comment]
DllNotFoundException吗？你试试UNITY_IPHONE的地方把visionOS的宏也加上。

**Reply by wangdinglu** [reply]
是Unity的，我试试谢谢

**Reply by wangdinglu** [reply]
试了增加UNITY_VISIONOS的宏下不会报DllNotFoundException了，但是依然无法启动，会报错在 PuertsDLL__GetApiLevel 部分

<img width="1393" alt="Screenshot 2023-12-12 at 20 17 22" src="https://github.com/Tencent/puerts/assets/27430103/8dabfbb1-c6cc-4eb7-9d91-8165b271710c">

**Reply by chexiongsheng** [reply]
如果是用__Internal去调用，为什么会触发dyld呢？另外关键信息是dyld那，你没截全

**Reply by chexiongsheng** [reply]
Apple Vision Pro用的是m2芯片，我感觉你用puerts的silicon支持会更合适些。

**Reply by Gary-Lty** [reply]
purest 版本2.0.4 在visionOS中仍然有这个问题，dyld[25988]: missing symbol called
![image](https://github.com/Tencent/puerts/assets/65580550/720c3524-65bc-47a2-9d98-20959ed4ed0c)

**Reply by Gary-Lty** [reply]
![image](https://github.com/Tencent/puerts/assets/65580550/24ed35dd-2f3f-4966-9656-aae9346e4354)

**Reply by chexiongsheng** [reply]
> ![image](https://private-user-images.githubusercontent.com/65580550/326373713-24ed35dd-2f3f-4966-9656-aae9346e4354.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0ODAyNzcsIm5iZiI6MTcxNDQ3OTk3NywicGF0aCI6Ii82NTU4MDU1MC8zMjYzNzM3MTMtMjRlZDM1ZGQtMmYzZi00OTY2LTk2NTYtYWFlOTM0NmU0MzU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDEyMjYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1NDYzNTE4YmE0MGRiNWEzZWFkMmMwZDliZDE1MzFhYjkyOGJjNjdlNGRkNWUyOWNlNmZjYTVmZTdjOWU4NDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aD-r_jmoQ16LkP9B1HK0xDcnXDOovn8_4F6pxYLt4Uk)
就是找不到库

**Reply by yosun** [reply]
> Apple Vision Pro uses the m2 chip, I think it would be more appropriate for you to use puerts' silicon support.

can you explain more on puerts' silicon support? is there a flag somewhere?
